src/software/thunderscope/robot_communication.py

```diff
@@ -23,26 +25,27 @@ def __init__(
         current_proto_unix_io: ProtoUnixIO,
         multicast_channel: str,
         interface: str,
-        disable_estop: bool,
-        estop_path: os.PathLike,
+        estop_mode: EstopMode,
+        estop_path: os.PathLike = None,
         estop_buadrate: int = 115200,
-    ) -> None:
+    ):
         """Initialize the communication with the robots
 
         :param current_proto_unix_io: the current proto unix io object
         :param multicast_channel: The multicast channel to use
         :param interface: The interface to use
-        :param disable_estop: whether to disable estop checks (ONLY FOR TESTING LOCALLY)
+        :param estop_mode: what estop mode we are running right now, of type EstopMode
         :param estop_path: The path to the estop
-        :param estop_baudrate: The baudrate of the estop
+        :param estop_buadrate: The baudrate of the estop
 
         """
         self.sequence_number = 0
         self.last_time = time.time()
         self.current_proto_unix_io = current_proto_unix_io
         self.multicast_channel = str(multicast_channel)
         self.interface = interface
-        self.disable_estop = disable_estop
+        self.estop_mode = estop_mode
+
```

Comment:
did you run formatting? there's a bunch of these extra new lines - might just be strange formatting on my end in the review mode.